strings-csharp/ReplaceSpecialCharactersInString/ReplaceSpecialCharactersInString/StringReplacer.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System.Text;
+using System.Text.RegularExpressions;
+
+
+namespace ReplaceSpecialCharactersInString
+{
+    public class StringReplacer
+    {
+        public static string ReplaceUsingStringReplace(string originalString,
+            char oldChar, char newChar)
+        {
+            return originalString.Replace(oldChar, newChar);
+        }
+
+
+        public static string ReplaceUsingStringBuilder(string originalString,
+            char oldChar, char newChar)
+        {
+            var sb = new StringBuilder(originalString);
+
+            sb.Replace(oldChar, newChar);
+
+            return sb.ToString();
+        }
+
+
+        public static string ReplaceUsingRegex(string originalString,
+            string oldChar, string newChar)
+        {
+            var pattern = $"[{Regex.Escape(oldChar)}]";
+
+            return Regex.Replace(originalString, pattern, newChar);
+        }
+
+
+        public static string ReplaceUsingSpan(string originalString,
+            char oldChar, char newChar)
+        {
+            var originalChars = originalString.ToCharArray();
+
+            var span = new Span<char>(originalChars);
+
+            span.Replace(oldChar, newChar);
+
+            return new string(span.ToArray());
+        }
+
+
+        public static string ReplaceUsingInefficientMultipleReplacementsStringReplace(
+            string originalString, char oldChar, char newChar)
+        {
+            var replacedString = originalString
```

Comment:
This is not the correct way to implement the library method. We can't hardcode the characters. For showing multiple String Replace what can't we call the first method "ReplaceUsingStringReplace" multiple times from caller method?